ln-gateway/src/bin/ln_gateway.rs

```diff
@@ -75,3 +78,17 @@ fn build_federation_client(
 
     Ok(Client::new(client_cfg, db, ctx))
 }
+
+/// Persist federation client cfg to [`gateway.json`] file
+fn save_federation_client_cfg(work_dir: PathBuf, client_cfg: &GatewayClientConfig) {
+    let path: PathBuf = work_dir.join("gateway.json");
+    if !Path::new(&path).is_file() {
+        debug!("Creating new gateway cfg file at {}", path.display());
+        let file = File::create(path).expect("Could not create gateway cfg file");
+        serde_json::to_writer_pretty(file, &client_cfg).expect("Could not write gateway cfg");
```

Comment:
Yeah, this just keeps existing persistence behavior. But, we don't even use the configs we write to file.
For now, this is only useful for debug, example, in case a gateway crashed and you wanted to inspect the config it was using...

I added a TODO above for us to think more about persistence of gateway config state, and how that leads to gateway recovery / migration